dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoDualWriterClient.java

```diff
@@ -174,74 +174,106 @@ public void setRange(int range) {
 
     @Override
     public Long append(final String key, final String value) {
+        return this.d_append(key, value).getResult();
+    }
+
+    @Override
+    public OperationResult<Long> d_append(final String key, final String value) {
         writeAsync(key, new Callable<OperationResult<Long>>() {
             @Override
             public OperationResult<Long> call() throws Exception {
-                return d_append(key, value);
+                return DynoDualWriterClient.super.d_append(key, value);
```

Comment:
Sounds weird that we are calling DynoDualWriterClient instead DynoDualWriterClient class. Wouldn't `this` not work here?